### ISSUE TITLE
Crash when mounting ship (MC 1.10.2 / 12.18.3.2239 / v0006-full)

### ISSUE THREAD

**Issue by andrewminer** [darkevilmac/archimedes-ships-plus#211]
I created a ship using only vanilla wood blocks and stairs along with a helm.  When I assembled, it seemed to work fine, but when I attempted to mount the ship, everything hung for about 30s and then crashed.  The log showed the following stacktrace:

```
Reported exception thrown!
net.minecraft.util.ReportedException: Ticking entity
	at net.minecraft.world.World.func_72939_s(World.java:1763)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1789)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
Caused by: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Byte
	at io.github.elytra.davincisvessels.common.entity.ShipCapabilities.canMove(ShipCapabilities.java:174)
	at io.github.elytra.davincisvessels.common.entity.EntityShip.handleControl(EntityShip.java:265)
	at io.github.elytra.movingworld.common.entity.EntityMovingWorld.handleServerUpdate(EntityMovingWorld.java:419)
	at io.github.elytra.davincisvessels.common.entity.EntityShip.handleServerUpdate(EntityShip.java:393)
	at io.github.elytra.movingworld.common.entity.EntityMovingWorld.func_70071_h_(EntityMovingWorld.java:362)
	at net.minecraft.world.World.func_72866_a(World.java:1964)
	at net.minecraft.world.World.func_72870_g(World.java:1934)
	at net.minecraft.world.World.func_72939_s(World.java:1750)
	... 10 more
```

**Comment by andrewminer** [comment]
To make things worse, when I log back in to my world, any time I move my view whirls about wildly making it impossible to see what's going on.  It stops a few seconds after I stop moving, but resumes immediately after I start moving again.  When I use F5 to switch my view, the same effect continues, but I can see that my players arms are waving about wildly.  The same is true of other mobs in my vicinity.

**Comment by Lurmey** [comment]
A little late on this but you may be able to use NBTExplorer to find and delete the ticking entity causing the crash, but I have no idea what's causing mobs and your own arms to flail around wildly. Sounds to me that perhaps you have an incompatible mod causing that but I don't know.

**Comment by darkevilmac** [comment]
Closed as 1.10 is no longer supported.
